modal/_output.py

```diff
@@ -498,6 +499,13 @@ async def _get_logs():
             elif isinstance(exc, StreamTerminatedError):
                 logger.debug("Stream closed. Retrying ...")
                 continue
+            elif isinstance(exc, socket.gaierror):
+                logger.debug("Lost connection. Retrying ...")
+                continue
+            elif isinstance(exc, AttributeError):
+                if "'NoneType' object has no attribute '_write_appdata'" in str(exc):
```

Comment:
It feels a little bit brittle to be parsing the string form of the exception like this, especially since `_write_appdata` is a private attribute, so a non-api-breaking upstream change could cause a regression (and this feels hard to write a good unit test for).

Also FYI, the `AttributeError` class has `.obj` and `.name` methods that would let you write this a little bit more elegantly (i.e. without checking the specific string) although the `.name` changing feels like more of a risk than the specific `AttributeError` syntax used by Python changing.

I wonder if there's a way we could make this more robust upstream in the client so that we don't cause the error to be triggered on this internal attribute lookup? Not exactly sure what that would look like.

If that sounds hard though (and I think it does), I am fine with merging a variant of this approach, perhaps with a TODO + ticket to make it more robust?

Reply:
Makes sense, changed the error catching and added a TODO!